lib/temporary-repository.js

```diff
@@ -29,6 +29,11 @@ function createTemporaryRepository(state) {
       name: state.temporaryName,
     },
   });
+
+  await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
we throttle mutating requests with a 3s timeout, but the 3s are awaiting before a request happened, not after. If the 1st recorded request is a mutating requests it all works fine. But if the 1st request is a `GET` or `HEAD` then we sometimes get a `404` because the repository did not yet replicate